src/wasm/wasm-type.cpp

```diff
@@ -513,24 +513,29 @@ FeatureSet Type::getFeatures() const {
   return getSingleFeatures(*this);
 }
 
-HeapType Type::getHeapType() const {
+static HeapType funcHeapType(HeapType::FuncKind),
+  externHeapType(HeapType::ExternKind), exnHeapType(HeapType::ExnKind),
+  anyHeapType(HeapType::AnyKind), eqHeapType(HeapType::EqKind),
+  i31HeapType(HeapType::I31Kind);
```

Comment:
Can you move these declarations into `Type::getHeapType` (but keep them static) so they don't pollute the global namespace in this file?

Reply:
This is more or less what I meant. I thought putting these into a class will not clutter the global namespace. I guess, as you suggested, `Type` is better than `TypeStore` for this, given that it already has all primitive types.

Reply:
Putting these inside the function would be less efficient, though? It would use atomic operations to create the object on the first call, and check constantly if it is the first call.

How about putting it in a `statics` local namespace? I pushed a commit with that.

Reply:
Oh, TIL!

Reply:
Um, aren't file static variables only visible within that file anyway? (I said something about the global namespace but nevermind that, it was not accurate; I guess I thought if it was within `Type::` it would be more cleaner.)

Reply:
Yeah, I was just hoping to hide them from the rest of wasm-type.cpp, but that's not worth the performance impact putting them in the function would have. I'm fine with the way it is now in the `statics` namespace 👍

Reply:
I'd be fine with any idea to improve this in a followup. But I do think it's reasonably encapsulated now.